12th-HW/Sources/App/ViewController.swift

```diff
@@ -0,0 +1,255 @@
+            shapeLayer.speed = 1.0
+            shapeLayer.timeOffset = 0.0
+            shapeLayer.beginTime = 0.0
+            let timeAfterPause = shapeLayer.convertTime(CACurrentMediaTime(), from: nil) - pausedTime
+            shapeLayer.beginTime = timeAfterPause
+            return
+        }
+        
+        let basicAnimation = CABasicAnimation(keyPath: "strokeEnd")
+        basicAnimation.toValue = 1
+        basicAnimation.duration = CFTimeInterval(timerDuration)
+        basicAnimation.fillMode = .forwards
+        basicAnimation.speed = 1.0
+        basicAnimation.isRemovedOnCompletion = true
+        shapeLayer.add(basicAnimation, forKey: "basicAnimation")
+    }
+    
+    private func stopAnimation() {
+        let pausedTime = shapeLayer.convertTime(CACurrentMediaTime(), from: nil)
+        shapeLayer.speed = 0
+        shapeLayer.timeOffset = pausedTime
+    }
+    
+    private func selectTypeOfTime() {
+        if  isWorkingTime {
+            timerLabel.text = "time to work"
+            timer.invalidate()
+            timerDuration = 10.0
+            playButtonIcon.isHidden = !playButtonIcon.isHidden
+            stopButtonIcon.isHidden = !stopButtonIcon.isHidden
+            startButton.backgroundColor = .systemOrange.withAlphaComponent(0.8)
+            startButton.setTitle("Start work", for: .normal)
+            roundLayer.strokeColor = UIColor.systemOrange.withAlphaComponent(0.4).cgColor
+            shapeLayer.strokeColor = UIColor.systemOrange.cgColor
+        } else {
+            timerLabel.text = "time to rest"
+            timer.invalidate()
+            timerDuration = 5.0
+            startButton.backgroundColor = .systemMint.withAlphaComponent(0.8)
+            startButton.setTitle("Start rest", for: .normal)
+            playButtonIcon.isHidden = !playButtonIcon.isHidden
+            stopButtonIcon.isHidden = !stopButtonIcon.isHidden
+            roundLayer.strokeColor = UIColor.systemMint.withAlphaComponent(0.4).cgColor
+            shapeLayer.strokeColor = UIColor.systemMint.cgColor
+        }
+    }
+    
+    // MARK: - Actions
+    
+    @objc private func startButtonPressed() {
+        isStarted = !isStarted
+        if isStarted {
+            timer = Timer.scheduledTimer(
+                timeInterval: 0.01,
+                target: self,
+                selector: #selector(timerTick),
+                userInfo: nil,
+                repeats: true)
+            changeButtonWhenStarting()
+            startAnimation()
+        } else {
+            timer.invalidate()
+            changeButtonWhenPaused()
+            stopAnimation()
+        }
+    }
+    
+    @objc private func timerTick() {
+        timerDuration -= 0.01
+        
+        // Change time format for last second
+        if timerDuration < 1.0 {
+            timerLabel.text = String(format: "%.1f", timerDuration)
+        } else {
+            timerLabel.text = String(format: "%.0f", timerDuration)
+        }
```

Comment:
```suggestion
        timerLabel.text = timerDuration < 1.0
        ? String(format: "%.1f", timerDuration)
        : String(format: "%.0f", timerDuration)
```
ну это просто рекомендация